sdks/python/apache_beam/transforms/display/display_data_test.py

```diff
@@ -0,0 +1,100 @@
+from __future__ import absolute_import
+
+from datetime import datetime
+import unittest
+
+import apache_beam as beam
+from apache_beam.transforms.display import HasDisplayData, DisplayData, DisplayDataItem
+
+
+class DisplayDataTest(unittest.TestCase):
+
+  def test_inheritance_ptransform(self):
+    class MyTransform(beam.PTransform):
+      pass
+
+    display_pt = MyTransform()
+    # PTransform inherits from HasDisplayData.
+    self.assertTrue(isinstance(display_pt, HasDisplayData))
+    self.assertEqual(display_pt.display_data(), {})
+
+  def test_inheritance_dofn(self):
+    class MyDoFn(beam.DoFn):
+      pass
+
+    display_dofn = MyDoFn()
+    self.assertTrue(isinstance(display_dofn, HasDisplayData))
+    self.assertEqual(display_dofn.display_data(), {})
+
+  def test_base_cases(self):
+    """ Tests basic display data cases (key:value, key:dict)
+    It does not test subcomponent inclusion
+    """
+    class MyDoFn(beam.DoFn):
+      def __init__(self, *args, **kwargs):
+        self.my_display_data = kwargs.get('display_data', None)
+
+      def process(self, context):
+        yield context.element + 1
+
+      def display_data(self):
+        return {'static_integer': 120,
+                'static_string': 'static me!',
+                'complex_url': {'value': 'github.com',
+                                'url': 'http://github.com',
+                                'label': 'The URL'},
+                'python_class': HasDisplayData,
+                'my_dd': self.my_display_data}
+
+    now = datetime.now()
+    fn = MyDoFn(display_data=now)
+    dd = DisplayData.create_from(fn)
+    dd_dicts = sorted([item.get_dict() for item in dd.items],
+                      key=lambda x: x['namespace']+x['key'])
+
+    nspace = '{}.{}'.format(fn.__module__, fn.__class__.__name__)
+    expected_items = [
+        {'url': 'http://github.com', 'namespace': nspace,
+         'value': 'github.com', 'label': 'The URL',
+         'key': 'complex_url', 'type': 'STRING'},
+        {'type': 'TIMESTAMP', 'namespace': nspace, 'key': 'my_dd',
+         'value': DisplayDataItem._format_value(now, 'TIMESTAMP')},
+        {'type': 'JAVA_CLASS', 'namespace': nspace,
+         'value': 'apache_beam.transforms.display.display_data.HasDisplayData',
+         'key': 'python_class'},
+        {'type': 'INTEGER', 'namespace': nspace,
+         'value': 120, 'key': 'static_integer'},
+        {'type': 'STRING', 'namespace': nspace,
+         'value': 'static me!', 'key': 'static_string'}]
+    expected_items = sorted(expected_items,
+                            key=lambda x: x['namespace']+x['key'])
+
+    self.assertEqual(dd_dicts, expected_items)
```

Comment:
In java, we found it useful to build a set of `DisplayDataMatchers` helpers to assert on specific display data properties. This was particularly valuable as we started adding display data on individual transforms. No need to add here, but consider it as you start adding display data around the SDK.